pkg/providers/instance/aksmachineinstance.go

```diff
@@ -0,0 +1,580 @@
+	}
+	osImageSubscriptionID, osImageResourceGroup, osImageGallery, osImageName, osImageVersion, err := parseVMImageID(vmImageID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse VM image ID %q: %w", vmImageID, err)
+	}
+	header := http.Header{}
+	header.Set("AKSHTTPCustomFeatures", "Microsoft.ContainerService/UseCustomizedOSImage")
+	header.Set("OSImageName", osImageName)                          // E.g. "2204gen2containerd"
+	header.Set("OSImageResourceGroup", osImageResourceGroup)        // E.g. "AKS-Ubuntu"
+	header.Set("OSImageSubscriptionID", osImageSubscriptionID)      // E.g., "10945678-1234-1234-1234-123456789012"
+	header.Set("OSImageGallery", osImageGallery)                    // E.g., "AKSUbuntu"
+	header.Set("OSImageVersion", osImageVersion)                    // E.g., "2022.10.03"
+	ctxWithHeader := context.WithValue(ctx, "vmimageid", vmImageID) // This line is really just for testing (see fake/aksmachinesapi.go). Azure-sdk-for-go is restrictive in extracting the header out.
+	ctxWithHeader = policy.WithHTTPHeader(ctxWithHeader, header)
+
+	poller, err := p.azClient.aksMachinesClient.BeginCreateOrUpdate(ctxWithHeader, p.clusterResourceGroup, p.clusterName, p.aksMachinesPoolName, aksMachineName, *aksMachineTemplate, nil)
+	if err != nil {
+		return nil, fmt.Errorf("aksMachine.BeginCreateOrUpdate for AKS machine %q failed: %w", aksMachineName, err)
+	}
+
+	// Get once after begin create to retrieve VMResourceID.
+	getResp, err := p.azClient.aksMachinesClient.Get(ctx, p.clusterResourceGroup, p.clusterName, p.aksMachinesPoolName, aksMachineName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get AKS machine instance %q once after begin creation: %w", aksMachineName, err)
+	}
+	gotAKSMachine := lo.ToPtr(getResp.Machine)
+	// Process what we got.
+	if gotAKSMachine.ID == nil {
+		return nil, fmt.Errorf("failed to get AKS machine instance %q once after begin creation: AKS machine ID is nil", aksMachineName)
+	}
+	if gotAKSMachine.Properties == nil {
+		return nil, fmt.Errorf("failed to get AKS machine instance %q once after begin creation: AKS machine properties is nil", aksMachineName)
+	}
+	if gotAKSMachine.Properties.ResourceID == nil {
+		return nil, fmt.Errorf("failed to get AKS machine instance %q once after begin creation: AKS machine VM resource ID is nil", aksMachineName)
+	}
+	if gotAKSMachine.Properties.Status == nil || gotAKSMachine.Properties.Status.CreationTimestamp == nil {
+		return nil, fmt.Errorf("failed to get AKS machine instance %q once after begin creation: AKS machine creation timestamp is nil", aksMachineName)
+	}
+	if gotAKSMachine.Properties.NodeImageVersion == nil {
+		return nil, fmt.Errorf("failed to get AKS machine instance %q once after begin creation: AKS machine node image version is nil", aksMachineName)
+	}
+	if gotAKSMachine.Properties.ProvisioningState != nil && lo.FromPtr(gotAKSMachine.Properties.ProvisioningState) == "Failed" {
```

Comment:
Shoul;d have consts for terminal ProvStates - I assume SDK already defines them - use them